### DISCUSSION TITLE
Flatpak Permissions Management

### DISCUSSION THREAD

**Discussion by candrapersada** [linuxmint/discussions#122] [Ideas]
The Linux Mint or Cinnamon allows changing what permissions have been granted to installed Flatpak applications.

**Comment by rcalixte** [comment]
Have you had a look at [Flatseal](https://flathub.org/apps/com.github.tchx84.Flatseal)? That already fits this need very well.

**Reply by eco-star** [reply]
The only "problem" here is Libadwaita.

**Reply by rcalixte** [reply]
Can you explain further? You can set global theming options with Flatseal as well (as long as your themes are accessible).

Are there any applications you've found that aren't theming well?

**Reply by eco-star** [reply]
When I install Flatseal and give access to the Mint-Themes at /usr/share/themes and restart Flatseal, it changes to white-mode. I also installed Mint-Y-Dark theme as Flatpak.

I think Flatseal is a libadwaita app and afaik it is not possible to set a custom theme to these type of apps.

**Reply by rcalixte** [reply]
I believe /usr is forbidden access from Flatpak applications so anything under that path would not be accessible. I use Flatseal to theme my Flatpak applications but I have a symlink for ~/.themes pointing to ~/.local/share/themes where I have all the themes I want for Flatpaks available. (I did this because of inconsistencies that exist with XDG variables for now.) Everything works so far, even Qt applications.

**Reply by eco-star** [reply]
Ah maybe you are right.
Now copied some themes from /usr/share/themes to ~/.themes (and also similar with icons) and set these configuration in Flatseal. After a restart it worked:
![image](https://github.com/linuxmint/discussions/assets/149502274/c1bbd4de-698f-4ef3-9abc-262a5130127e)

Thanks a lot @rcalixte :)

**Reply by rcalixte** [reply]
The only thing I have configured differently is that I have `xdg-data/icons:ro` and `xdg-data/themes:ro` for the filesystem permissions (which is why I use the symlink for themes) and I set `XCURSOR_THEME` in the environment variables so that it would also be consistent across Flatpaks applications and local ones.

**Reply by eco-star** [reply]
Do you have multiple users on your PC or why do you have all your themes in ~/.local/share/themes? Oh yes, this is a good advise with the cursor theme. Thank you very much for your help! :)

I think this configuration should be default for all flatpak-programs to create more consistency.

**Reply by rcalixte** [reply]
> Do you have multiple users on your PC or why do you have all your themes in ~/.local/share/themes? Oh yes, this is a good advise with the cursor theme. Thank you very much for your help! :)

Glad to be helpful! I compile my themes for now (I'm working on creating my own) and the themes downloaded through the Themes in the settings default to one of the theme locations (I can't remember which). It's easy enough as a standard location.


> I think this configuration should be default for all flatpak-programs to create more consistency.

Agreed, though I do have some systems configured with different themes between Flatpaks and non-Flatpak apps. I've seen a lot of different takes on how users theme their Flatpaks, including installing the themes via Flatpak. I've just found what works for me and is easy enough to configure.

**Reply by eco-star** [reply]
> Glad to be helpful! I compile my themes for now (I'm working on creating my own) and the themes downloaded through the Themes in the settings default to one of the theme locations (I can't remember which). It's easy enough as a standard location.

This sounds interesting. Maybe you can share your own theme when it is ready.

> Agreed, though I do have some systems configured with different themes between Flatpaks and non-Flatpak apps. I've seen a lot of different takes on how users theme their Flatpaks, including installing the themes via Flatpak. I've just found what works for me and is easy enough to configure.

Yes, this was the way I tried to theme my apps by installing the Mint-Y-Dark-Theme from Flathub. I hoped this would set the theme of all flatpaks automatically, but I was wrong. But you are right. This is really an easy enough way for linux enthusiasts.

**Reply by rcalixte** [reply]
> Maybe you can share your own theme when it is ready.

This is my plan. It will be available through the Spices website and the Themes menu option once completed. Hopefully next year I can get that pull request open. 😅
